### ISSUE TITLE
[Feature Request] Add "Exit Reader View" Button to Quickly Return to Landing Page

### ISSUE THREAD

**Issue by bmmmm** [FreshRSS/FreshRSS#7011]
**Is your feature request related to a problem? Please describe.**  
Not exactly a problem, but a usability improvement. While immersed in reading feeds in the reader view, it’s sometimes cumbersome to navigate back to the landing page or main overview of FreshRSS.

**Describe the solution you’d like**  
I propose adding a floating button (positioned at the bottom, right, or left side of the interface) that allows users to exit the reader view and return directly to the landing page or main feed overview. This would streamline the transition and enhance usability.  

**Describe alternatives you’ve considered**  
Currently, I manually click the title bar (URL) and remove the trailing path to navigate back to the landing page. 

Example visualization:  
![Example Visualization](https://github.com/user-attachments/assets/e74e5961-ed6d-43b5-af5a-c62c0914a7ec)

**Additional context**  
The left and right margins of the reader interface remain largely unused. It might be interesting to explore using these areas for navigation or quick actions.  I am thinking of "using" them for navigation / modification could be cool? 😊

BIG THANKS to the project team!!!1💖💖

**Comment by Abdallah-Abdaljalil** [comment]
Hi @bmmmm, I'm Interested in working on this issue. I would highly appreciate it if you could assign this issue to me.

**Comment by patman62** [comment]
Doesn't clicking the Normal View (or Global View) take you back to where you were before? As a relatively newcomer to FreshRSS the UX seems a little intimidating to noobs and this proposal would not improve that. Of course, if it was opt-in in the display settings that would be ok.

**Comment by Abdallah-Abdaljalil** [comment]
> Doesn't clicking the Normal View (or Global View) take you back to where you were before? As a relatively newcomer to FreshRSS the UX seems a little intimidating to noobs and this proposal would not improve that. Of course, if it was opt-in in the display settings that would be ok.

From what I understand, when the user is in reading mode, it shows all the content under a section (Like Planet GNOME), I guess. So if you were to scroll far whilst reading and you would like to return to the main page, it would be easier to have a button that takes you there straight away that is located on the open space on the left and right. Because, from my experience, the user has to go back to the top to get back to normal or global view or whatever. I don't know if I am making sense here or if I am wrong, but I would appreciate a response to indicate what you think.

**Comment by bmmmm** [comment]
hello everyone,
@patman62, personally, could you write down and specify your points in a separate post. so we can have a discussion on them. We might can contribute to improve the documentation or can even make some suggestions to improve the UI. 

> Doesn't clicking the Normal View (or Global View) take you back to where you were before?

If you are in reader mode ([book symbol](https://demo.freshrss.org/themes/icons/view-reader.svg)), the menu bar and top bar are floating "away" as you are scrolling "down" your articles. If you scroll up, to reach the bars or home button, you have to scroll aaaaaall the way up >.< 

@Abdallah-Abdaljalil - there are similar buttons in place while being in list view and having the source overview the left side:
<img width="345" alt="image" src="https://github.com/user-attachments/assets/eca0883f-06eb-41b4-9484-577609ecae32">
And sorry, I have no permissions to assign you the issue 😕

**Comment by patman62** [comment]
Yes, but if you use the 'Fixed Nav Menu' plugin the buttons never leave the top of the screen as you scroll down. I don't understand the default behavior of scrolling the search box and the buttons off screen as the 'content' is scrolled (I can't think of any other apps that do that) but I'm glad there is a plugin that partly addresses it.

**Comment by Frenzie** [comment]
Many do, but this is a somewhat older implementation that doesn't show it when scrolling up. Let me see if I can whip up something in a few minutes when I get back.

**Comment by bmmmm** [comment]
Quick update after a while, since #7029 was reverted in #7064 and this is still open.

While looking into it, I noticed that most of this may already exist:
`app/layout/nav_entries.phtml` provides prev / **up** / next buttons, fixed to the
bottom left, and the `.up` handler in `p/scripts/main.js` scrolls back to the top.
It is already opt-out through `show_nav_buttons`.

It is simply not rendered in the reader view — only in
`app/views/index/normal.phtml:216`. On top of that, `frss.css:1939` hides it
whenever the sidebar is not visible:

```css
.aside:not(.visible) ~ #nav_entries { display: none; }
```

which always applies in the reader view, because of `.reader .aside { display: none }`.

Why this matters in the reader view in particular: on my instance 20 articles
produce a page of about 14000px against a viewport of about 990px, and
`auto_load_more` keeps extending it while scrolling — so there is no practical
way back to the top.

There is already a precedent for showing the widget without a sidebar: the mobile
media query (`frss.css:2865`) decouples it with `display: flex !important; width: 100%`.

Would a small PR be welcome that renders `nav_entries` in the reader view and adds
the matching CSS exception? It would not touch the header, so it should not
interfere with #6047. I would verify the themes that define their own `#nav_entries`
rules.

**Comment by bmmmm** [comment]
Update on my own request, in case it saves someone else the exploration.

I prototyped the obvious approach locally. The reading view already has almost
everything: `app/layout/nav_entries.phtml` exists, `init_nav_entries()` already
runs there, and `show_nav_buttons` already covers the opt-out. The widget is
simply never rendered by `reader.phtml`, and `.aside:not(.visible) ~
#nav_entries` would hide it anyway, because `.reader .aside` is `display: none`.
Roughly 20 lines, no JavaScript change. Themes are not a complication — they all
load `_frss.css` and only override colours.

Two reasons I am not turning it into a PR:

**It does not solve what I actually asked for.** The `up` button scrolls back to
the top, where the toolbar is reachable again — useful, but not a one-click exit
from the reading view.

**It overlaps the article.** In the normal view the widget sits in the sidebar
column and covers nothing. The reading view has no sidebar, so a `position:
fixed` widget lands on top of the centred text — always at *Full width*, and
below roughly 1390px viewport at *Wide*. The article card has no horizontal
padding, so the overlap hits body text. Which means the "the margins are largely
unused" assumption in my original post does not hold for the wider settings.

Given that #7029 and #7063 both ran into UI regressions at exactly this spot, a
third core attempt does not look like the right move. So I built the exit button
as an extension instead, which leaves the core UI alone:

https://github.com/bmmmm/xExtension-ExitReaderView

It adds a small floating control in the reading view that returns to the normal
view in one click, keeping the active feed, category, tag or search. AGPL-3.0,
English and German included.

Happy for this to be closed, or to leave it open in case the topic returns with
#6047.

**Comment by Frenzie** [comment]
> since #7029 was reverted

I didn't intend it to merged, just to be discussed. It didn't so much run into issues as that it simply wasn't finished.

Or as I wrote in the OP:

> This is just a quick poc, but it shows a possible approach.
